homeassistant/components/dlink/switch.py

```diff
@@ -59,21 +64,27 @@ def extra_state_attributes(self) -> dict[str, Any]:
     @property
     def is_on(self) -> bool:
         """Return true if switch is on."""
-        return self.data.state == "ON"
+        return self.coordinator.data.get("state") == "ON"
 
-    def turn_on(self, **kwargs: Any) -> None:
+    async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the switch on."""
-        self.data.smartplug.state = "ON"
 
-    def turn_off(self, **kwargs: Any) -> None:
+        def _turn_on(smartplug) -> None:
+            smartplug.state = "ON"
+
+        await self.hass.async_add_executor_job(_turn_on, self.coordinator.smartplug)
+        await self.coordinator.async_request_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the switch off."""
-        self.data.smartplug.state = "OFF"
 
-    def update(self) -> None:
-        """Get the latest data from the smart plug and updates the states."""
-        self.data.update()
+        def _turn_off(smartplug) -> None:
+            smartplug.state = "OFF"
+
+        await self.hass.async_add_executor_job(_turn_off, self.coordinator.smartplug)
+        await self.coordinator.async_request_refresh()
```

Comment:
These def -> async def aren't needed for the coordinator right? Can you maybe leave that for a follow-up PR?

Reply:
I think they are: if you don't call the `async_request_refresh` the switch state won't change until the next polling, right?

Reply:
We also don't need separate attributes from the coordinator for the entities. Just pass the smart plug object.